dotnet/samples/Demos/AotCompatibility.TestApp/README.md

```diff
@@ -0,0 +1,50 @@
+   
+To run the tests in Native-AOT mode, publish the app in the same way as above, and then run it with the `-tests` argument: `.\bin\Release\net8.0\win-x64\publish\AotCompatibility.TestApp.exe -tests`.
+
+## Samples
+Most of the samples don't require any additional setup, and can be run as is. However, some of them might require additional configuration.
+
+### 1. [ONNX Chat Completion Service](./Samples/OnnxChatCompletionSamples.cs)
+To configure the sample, you need to download the ONNX model from the Hugging Face repository. Go to a directory of your choice where the model should be downloaded and run the following command:
+```powershell
+git clone https://huggingface.co/microsoft/Phi-3-mini-4k-instruct-onnx
+```
+
+> [!IMPORTANT]
+The `Phi-3` model may be too large to download using the `git clone` command unless you have the [git-lfs extension](https://git-lfs.com/) installed. 
+You might need to download it manually using the following link: [Phi-3-Mini-4k CPU](https://huggingface.co/microsoft/Phi-3-mini-4k-instruct-onnx/resolve/main/cpu_and_mobile/cpu-int4-rtn-block-32/phi3-mini-4k-instruct-cpu-int4-rtn-block-32.onnx.data?download=true) (approximately 2.7 GB).
+
+After downloading the model, you need to configure the sample by setting the `Onnx:ModelPath` and `Onnx:ModelId` secrets. 
+The `Onnx:ModelPath` should point to the directory where the model was downloaded, and the `Onnx:ModelId` should be set to `phi-3`.
+The secrets can be set using [Secret Manager](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets#secret-manager) in the following way:
+```powershell
+dotnet user-secrets set "Onnx:ModelId" "phi-3"
+dotnet user-secrets set "Onnx:ModelPath" "C:\path\to\huggingface\Phi-3-mini-4k-instruct-onnx\cpu_and_mobile\cpu-int4-rtn-block-32" 
+```
+
+### AOT Compatibility
+At the moment, the following Semantic Kernel packages are AOT compatible:
+
+| Package                   | AOT compatible |  
+|--------------------------|----------------|  
+| SemanticKernel.Abstractions | ✔️              |  
+| SemanticKernel.Core         | ✔️              |  
+| Connectors.Onnx            | ✔️              |  
+
+Other packages are not AOT compatible yet, but we plan to make them compatible in the future.
+
+### Known Issues
+#### 1. KernelFunction JSON Schema
+Semantic Kernel uses System.Text.Json (STJ) v8 to generate JSON schemas for the parameters and return types of kernel plugin functions. 
+However, because STJ v8 uses reflection to resolve certain metadata, such as nullability annotations and constructor parameters, 
+it produces an incorrect JSON Schema in Native-AOT scenarios. For more details, see the following issue: [Incorrect type schema ...](https://github.com/eiriktsarpalis/stj-schema-mapper/issues/7).
+This issue can be worked around by disabling the `IlcTrimMetadata` property in the application's project; however, this may increase the size of the application.
```

Comment:
```suggestion
> [!NOTE]
> This issue can be worked around by disabling the `IlcTrimMetadata` property in the application's project; however, this may increase the size of the application.
```

Reply:
I think it reads well without the extra emphasis on the workaround. The preceding sentences gradually introduce the context, and the last sentence about the workaround logically concludes the issue description.